sstables/sstable_set.cc

```diff
@@ -61,10 +61,11 @@ std::ostream& operator<<(std::ostream& os, const sstables::sstable_run& run) {
     return os;
 }
 
-sstable_set::sstable_set(std::unique_ptr<sstable_set_impl> impl, schema_ptr s, lw_shared_ptr<sstable_list> all)
+sstable_set::sstable_set(std::unique_ptr<sstable_set_impl> impl, schema_ptr s)
         : _impl(std::move(impl))
         , _schema(std::move(s))
-        , _all(std::move(all)) {
+        , _all(make_lw_shared<sstable_list>()) {
```

Comment:
why the explicit initializer?
the implicit default constructor is just what you need.

Reply:
No, the default constructor creates a null pointer. This creates a pointer looking at a unique, empty list.